lib/ddtrace/utils.rb

```diff
@@ -29,8 +33,24 @@ def self.truncate(value, size, omission = '...')
       string.slice(0, size - omission.size) + omission
     end
 
-    private_class_method :reset!, :was_forked?
+    def utf8_encode(str, options = {})
+      str = str.to_s
+
+      if str.encoding?(Encoding::UTF_8)
+        str
+      elsif options[:binary]
+        # This option is useful for "gracefully" displaying binary data that
+        # often contains text such as marshalled objects
+        str.encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
+      else
+        str.encode(Encoding::UTF_8)
+      end
+    rescue => e
+      Tracer.log.error("Error encoding string in UTF-8: #{e}")
+
+      options[:placeholder] || STRING_PLACEHOLDER
+    end
 
-    reset!
+    STRING_PLACEHOLDER = ''.freeze
```

Comment:
Minor, but I think maybe this could be defined at the top of the file? (As is more conventional.)